invenio_edugain/utils.py

```diff
@@ -57,6 +62,65 @@ def get_idp_data_dict() -> dict:
     }
 
 
+def load_mdstore(
+    metadata_xml_location: PathLike | str,
+    cert_location: PathLike | str | None = None,
+    fingerprint_sha256: str | None = None,
+    *,
+    http_client_timeout: int = 30,
+) -> MetadataStore:
+    """Load a pysaml2 MetadataStore from given path/url.
+
+    When loading metadata from url, requires a certificate to check validity of metadata.
+    When loading that certificate from url, requires a fingerprint to check validity of certificate.
+    """
+    mds = MetadataStore(None, Config(), http_client_timeout=http_client_timeout)
+
+    if isinstance(metadata_xml_location, str) and validators.url(metadata_xml_location):
```

Comment:
can you please simplify the if else construct. there are to many levels. it is hard to grep which situation is handled how. 

you could move the `if isinstance(cert_location, str) and validators.url(cert_location):` case in a separate function

Reply:
fair enough

a function would have to take a buttload of arguments
maybe better to do early returns for the `easy` cases?

I'll figure something out